packages/playwright-test/src/reporters/merge.ts

```diff
@@ -260,16 +259,29 @@ function mergeConfigs(to: JsonConfig, from: JsonConfig): JsonConfig {
 }
 
 function mergeEndEvents(endEvents: JsonEvent[]): JsonEvent {
-  const result: FullResult = { status: 'passed' };
+  let startTime = endEvents.length ? 10000000000000 : Date.now();
+  let endTime = endEvents.length ? 0 : Date.now();
+  let status: JsonFullResult['status'] = 'passed';
+  let duration: number = 0;
+
   for (const event of endEvents) {
-    const shardResult: FullResult = event.params.result;
+    const shardResult: JsonFullResult = event.params.result;
     if (shardResult.status === 'failed')
-      result.status = 'failed';
-    else if (shardResult.status === 'timedout' && result.status !== 'failed')
-      result.status = 'timedout';
-    else if (shardResult.status === 'interrupted' && result.status !== 'failed' && result.status !== 'timedout')
-      result.status = 'interrupted';
+      status = 'failed';
+    else if (shardResult.status === 'timedout' && status !== 'failed')
+      status = 'timedout';
+    else if (shardResult.status === 'interrupted' && status !== 'failed' && status !== 'timedout')
+      status = 'interrupted';
+    startTime = Math.min(startTime, shardResult.startTime);
+    endTime = Math.max(endTime, shardResult.endTime);
+    duration += shardResult.duration;
   }
+  const result: JsonFullResult = {
+    status,
+    startTime,
+    endTime,
```

Comment:
Perhaps we should set `endTime` to `now()`, but `startTime` to `endTime - duration`?

Reply:
Can't do math on monotonic vs wall time.